src/NodaTime/LocalInstant.cs

```diff
@@ -187,7 +187,7 @@ internal Instant SafeMinus(Offset offset)
         /// <c>true</c> if the specified <see cref="System.Object"/> is equal to this instance;
         /// otherwise, <c>false</c>.
         /// </returns>
-        public override bool Equals(object obj) => obj is LocalInstant && Equals((LocalInstant)obj);
+        public override bool Equals(object? obj) => obj is LocalInstant other && Equals(other);
```

Comment:
Not that it matters too much, but we seem to be writing `Equals(object?)` three different ways:

`=> obj is LocalInstant other && Equals(other);`
`=> obj is LocalInstant other && this == other;`
`=> Equals(obj as LocalInstant)`

Is there any reason to prefer one over the others? If not, can we at least make them consistent?

Reply:
The first is the most common one for value types.

The second is a very slight (perhaps negligible) optimization if I happen to spot that `Equals` calls `==` directly.

The third is the simplest way of doing the first but for reference types.

So I don't mind 1 and 3 existing. Perhaps we should standardize between 1 and 2 though.